ansible/deploy.py

```diff
@@ -0,0 +1,162 @@
+  env['ANSIBLE_VAULT_IDENTITY'] = host
+  env['ANSIBLE_SSH_PIPELINING'] = 'true'
+  env['ANSIBLE_VAULT_PASSWORD_FILE'] = './scripts/bitwarden-vault-pass.py'
+
+  arguments = [
+    'ansible-playbook',
+    '--inventory',
+    './hosts',
+    '--diff',
+    "--limit",
+    host,
+    '--extra-vars',
+    f"playbook_revision={revision}",
+  ]
+
+  # Special scenario: When restore-backup.yml is run on staging, it needs to
+  # access the production Vault secrets, so this adds the call for that to
+  # the environment passed to ansible-playbook
+  if playbook == 'playbooks/restore-backup.yml':
+    arguments.append('--vault-id production@prompt')
+
+  if check:
+    arguments.append('--check')
+
+  arguments.append(playbook)
+
+  if not check:
+    notify_deploy_start(playbook, host, user, branch, revision)
+
+  print("Running the following playbook:")
+  print(' '.join(arguments))
+
+  try:
+    subprocess.run(
+      arguments,
+      check=True,
+      env=env
+    )
+    if not check:
+      notify_deploy_succes(playbook, host, branch, revision)
+
+  except CalledProcessError:
+    if not check:
+      notify_deploy_fail(playbook, host, branch, revision)
+
+
+def current_branch_name():
+  repo = git.Repo('..')
+  return repo.active_branch
+
+
+def current_git_revision():
+  repo = git.Repo('..')
+  return repo.commit().hexsha
+
+
+def notify_deploy_start(playbook, host, user, git_branch, git_revision):
+  discord_notify(
+    f"*Deployment of playbook {playbook} in {host} environment started by {user}*\n" +
+    f"_(branch: {git_branch} - revision \"{git_revision}\")_",
+    ':construction:',
+    '#46c4ff'
+  )
+
+def notify_deploy_succes(playbook, host, git_branch, git_revision):
+  discord_notify(
+    f"*Deployment of playbook {playbook} in {host} environment succesfully completed*\n" +
+    f"_(branch: {git_branch} - revision \"{git_revision}\")_",
+    ':construction:',
+    'good'
+  )
+
+def notify_deploy_failure(playbook, host, git_branch, git_revision):
+  discord_notify(
+    f"*Deployment of playbook {playbook} in {host} environment FAILED!*\n" +
+    f"_(branch: {git_branch} - revision \"{git_revision}\")_",
+    ':exclamation:',
+    'danger'
+  )
+
+def discord_notify(message, icon, color):
+  url = get_slack_webhook().strip()
+
+  data = {
+    'username': 'Ansible',
+    'attachments': [
+      {
+        'color': color,
+        'mrkdwn_in': [
+          'text',
+          'fields',
+        ],
+        'text': message,
+      },
+    ],
+    'icon_emoji': icon,
+  }
+
+  r = requests.post(
+    url,
+    json=data,
+  )
+
+  if r.status_code != 200:
+    raise click.ClickException(f"Discord failed with statuscode {r.status_code} and message {r.text}")
+
+def verify_on_latest_master(host):
+  repo = git.Repo('..')
+
+  # not sure why str() needs to be called
+  if host == 'production' and str(repo.active_branch) != 'master':
+    raise click.ClickException('Please do not deploy anything but master to production')
```

Comment:
We probably want a way to disable this check if we ever need to deploy a non-master branch in case of trouble.